Examples/ObjectiveC/WebAPIDataExample.m

```diff
@@ -48,48 +48,32 @@ - (void)addItemsToMap:(NSArray *)features {
     // Use MGLCircleStyleLayer to represent the points with simple circles.
     // In this case, we can use style functions to gradually change properties between zoom level 2 and 7: the circle opacity from 50% to 100% and the circle radius from 2pt to 3pt.
     MGLCircleStyleLayer *circles = [[MGLCircleStyleLayer alloc] initWithIdentifier:@"lighthouse-circles" source:source];
-    circles.circleColor = [MGLStyleValue valueWithRawValue:lighthouseColor];
-    circles.circleOpacity = [MGLStyleValue valueWithInterpolationMode:MGLInterpolationModeExponential
-        cameraStops:@{
-            @2: [MGLStyleValue valueWithRawValue:@0.5],
-            @7: [MGLStyleValue valueWithRawValue:@1.0]
-        }
-        options:@{MGLStyleFunctionOptionDefaultValue:[MGLStyleValue valueWithRawValue:@0.75]}];
-                             
-    circles.circleRadius = [MGLStyleValue valueWithInterpolationMode:MGLInterpolationModeInterval
-        cameraStops:@{
-            @2: [MGLStyleValue valueWithRawValue:@2],
-            @7: [MGLStyleValue valueWithRawValue:@3]
-        }
-        options:@{MGLStyleFunctionOptionDefaultValue:@1}];
-    
+    circles.circleColor = [NSExpression expressionForConstantValue:lighthouseColor];
+    circles.circleOpacity = [NSExpression expressionWithFormat:@"mgl_interpolate:withCurveType:parameters:stops:($zoomLevel, 'linear', nil, %@)",
+                             @{@2: @0.5, @7: @1.0 }];
+    circles.circleRadius = [NSExpression mgl_expressionForSteppingExpression: NSExpression.zoomLevelVariableExpression
+                                                              fromExpression:[NSExpression expressionForConstantValue:@1]
+                                                                       stops:[NSExpression expressionForConstantValue: @{@2: @2, @7: @3}]];
+
     // Use MGLSymbolStyleLayer for more complex styling of points including custom icons and text rendering.
     MGLSymbolStyleLayer *symbols = [[MGLSymbolStyleLayer alloc] initWithIdentifier:@"lighthouse-symbols" source:source];
-    symbols.iconImageName = [MGLStyleValue valueWithRawValue:@"lighthouse"];
-    symbols.iconScale = [MGLStyleValue valueWithRawValue:@0.5];
-    symbols.iconOpacity = [MGLStyleValue valueWithInterpolationMode:MGLInterpolationModeExponential
-        cameraStops:@{
-            @5.9: [MGLStyleValue valueWithRawValue:@0],
-            @6: [MGLStyleValue valueWithRawValue:@1],
-        }
-        options:nil];
-    symbols.iconHaloColor = [MGLStyleValue valueWithRawValue:[[UIColor whiteColor] colorWithAlphaComponent:0.5]];
-    symbols.iconHaloWidth = [MGLStyleValue valueWithRawValue:@1];
+    symbols.iconImageName = [NSExpression expressionForConstantValue:@"lighthouse"];
+    symbols.iconScale = [NSExpression expressionForConstantValue:@0.5];
+    symbols.iconOpacity = [NSExpression expressionWithFormat:@"mgl_interpolate:withCurveType:parameters:stops:($zoomLevel, 'linear', nil, %@)",
```

Comment:
Same with this one - additional clarification would help users understand what is happening here.